packages/components/src/components/segmented-control/segmented-control-item/segmented-control-item-styles.ts

```diff
@@ -64,32 +65,49 @@ export const getComponentCss = (
         display: 'block',
         height: '100%',
         width: '100%',
-        padding: `${pxToRemWithUnit(12)} ${ITEM_PADDING}`,
+        padding: hasIcon ? `13px ${ITEM_PADDING} 13px 13px` : `13px ${ITEM_PADDING}`,
         margin: 0,
-        border: 0,
-        background: backgroundColor,
+        border: `${borderWidthBase} solid ${borderColor}`,
+        outline: 0,
+        backgroundColor: 'transparent',
         color: buttonColor,
         ...textSmallStyle,
-        ...getFocusJssStyle({ color: primaryColor }),
+        overflowWrap: 'normal',
+        position: 'relative',
+        borderRadius: borderRadiusSmall,
+        '&::before': {
+          content: '""',
+          position: 'absolute',
+          ...getInsetJssStyle(-5),
+          border: `${borderWidthBase} solid transparent`,
+          borderRadius: '7px',
+        },
+        '&:focus::before': {
+          borderColor: focusColor,
+        },
+        '&:focus:not(:focus-visible)::before': {
+          borderColor: 'transparent',
+        },
         ...(isDisabled
           ? {
               cursor: 'not-allowed',
             }
           : {
               cursor: 'pointer',
-              ...(!isSelected &&
-                hoverMediaQuery({
-                  transition: getTransition('background-color'),
-                  '&:hover': {
-                    background: contrastLowColor,
-                  },
-                })),
+              ...hoverMediaQuery({
```

Comment:
is `isSelected` not needed anymore?

Reply:
@stephanschroeter it's needed indeed (not necessarily, but it would be more correct) The original idea of designers was that the hover state applies also to selected element, therefore I removed isSelected here. But then we changed the design, and now it would be correct to have isSelected again here.

Reply:
Fixed it